src/language-js/needs-parens.js

```diff
@@ -386,31 +386,8 @@ function needsParens(path, options) {
       }
 
     case "SequenceExpression":
-      switch (parent.type) {
-        case "ReturnStatement":
-          return false;
-
-        case "ForStatement":
-          // Although parentheses wouldn't hurt around sequence
-          // expressions in the head of for loops, traditional style
-          // dictates that e.g. i++, j++ should not be wrapped with
-          // parentheses.
-          return false;
-
-        case "ExpressionStatement":
-          return name !== "expression";
```

Comment:
What's the history behind this `case`? Someone wrote this for a reason I suppose. What was this reason?

Reply:
https://github.com/benjamn/ast-types/commit/72ce1cd3b6125e3e73c00279e055ecec80318bba#diff-cba79daef5f373bdd772d7998a1fe3cb seems to be the first commit related to this.

After that, it was moved to recast via https://github.com/benjamn/recast/commit/47700c22a3c13b322a9c091af592dcadec9c1aab#diff-55af5741c941ea6cde2dee31b5b8d159,

and it seems to have moved to Prettier via https://github.com/prettier/prettier/commit/9b4535e9f8a7040f2fbc65b22c9713632f3731be#diff-db385a780be1944fa983e81c7e7b720aR253-R272.